### DISCUSSION TITLE
配置 Node.js 开发环境——使用 Atom — 码志

### DISCUSSION THREAD

**Discussion by mzlogin** [mzlogin/blog-comments#218] [Announcements]
https://mazhuang.org/2016/04/18/develop-nodejs-with-atom/

**Comment by AriAdler** [comment]
本地安装 中的
git clone git@github.com:tststs/atom-ternjs.git
应改为
git clone https://github.com/tststs/atom-ternjs.git

**Comment by mzlogin** [comment]
我在我用到的几台机器上试过用 git@github.com:tststs/atom-ternjs.git 这样的地址，都是可以的，请问你那边 git clone git@github.com:tststs/atom-ternjs.git 是遇到什么问题了呢？

**Comment by AriAdler** [comment]
我的环境
fedora 25
linux kernel 4.11.8
atom 1.18.0 x64

报错如下
Warning: Permanently added the RSA host key for IP address '192.30.255.113' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

出现该异常的原因可能是 我在没有配置ssh的情况下，使用ssh连接操作github库
github推荐使用https方式进行连接，我建议 将 原文中
本地安装 中的
git clone git@github.com:tststs/atom-ternjs.git
应改为
git clone https://github.com/tststs/atom-ternjs.git
这样无需配置 ssh 方便简洁

**Comment by mzlogin** [comment]
噢噢~确实我试的几台机器都是配置过 ssh 的 key 的。这样的话确实使用 https 会更普适。

文中已修改，感谢反馈！

**Comment by YuetTusi** [comment]
我安装了script插件一直报错，找不到某个包，请问script插件安装要不要做一些配置？我用它开发node，谢谢

**Comment by mzlogin** [comment]
@YuetTusi 我现在基本不怎么用 Atom 啦……你直接到 script 插件的项目去提问题看看吧。

**Comment by MemoryReload** [comment]
不提交代码的情况下用https传输协议，或者git传输协议都没差别。如果提交代码，https每次输入密码，那时候你才知道什么是不方便。用git不会配置ssh认证，还说不方便，这叫不科学……

**Comment by mzlogin** [comment]
> @MemoryReload
> 不提交代码的情况下用https传输协议，或者git传输协议都没差别。如果提交代码，https每次输入密码，那时候你才知道什么是不方便。用git不会配置ssh认证，还说不方便，这叫不科学……

是指评论中提到的 https 还是 ssh 的讨论么？哈哈 嗯 你的说法没毛病，需要提交代码用 ssh 当然是方便得多。不过使用 Atom 并用 git 来拷贝安装插件的，不需要向插件提交代码，也不一定是 GitHub 用户，所以我觉得这种写操作步骤类的文里，用 https 普适性比较强。

**Comment by MemoryReload** [comment]
我的意思是，这种讨论毫无意义，也不是我装逼就说git协议的连接方式就好。但是，作为一个开发者，不配置ssh是比较少见的，毕竟，永远不用输入密码，要比输入密码这个来的简单。一劳永逸对于一个慵懒的程序员来说，是更好的选择。“Configure once，use anytime！”，哈哈哈～如果你这个是给小白看的，修改成https的会更好。
